src/lib/util/view.h

```diff
@@ -0,0 +1,154 @@
+// license:BSD-3-Clause
+// copyright-holders:Nathan Woods
+/******************************************************************************
+
+	view.h
+
+	STL container for a view
+
+***************************************************************************/
+
+#pragma once
+
+#ifndef __VIEW_H__
+#define __VIEW_H__
+
+#include <cstddef>
+
+namespace util {
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+template<typename T >
+class view
+{
+public:
+	typedef std::ptrdiff_t difference_type;
+	typedef std::size_t size_type;
+	typedef T value_type;
+	typedef T &reference;
+	typedef const T &const_reference;
+	typedef T *pointer;
+
+	class iterator
+	{
+	public:
+		typedef std::ptrdiff_t difference_type;
+		typedef T value_type;
+		typedef T &reference;
+		typedef T *pointer;
+
+		typedef std::random_access_iterator_tag iterator_category;
+
+		iterator(T *ptr) : m_ptr(ptr) { }
+		iterator(const iterator &that) : m_ptr(that) { }
```

Comment:
You can use `= default` or not declare this since it just does a memberwise copy.  Same for the assignment operator.